lib/pool.js

```diff
@@ -159,11 +167,26 @@ class Pool extends EventEmitter {
 
       if (!client) {
         if (!this[kConnections] || this[kClients].length < this[kConnections]) {
-          client = new Client(this[kUrl], this[kOptions])
+          const options = { ...this[kOptions] }
+          if (!options.tls || (options.tls.reuseSessions !== false)) {
```

Comment:
`options.tls.reuseSessions` is a flag to disable the session reuse in a conn. pool. I thought of it as a safe mechanism should it ever be needed. The sessions are reused within a pool by default.